elasticache/elasticache.tf

```diff
@@ -1,15 +1,23 @@
-# Non encrypted basic cluster, for caching
-resource "aws_elasticache_cluster" "main" {
-  cluster_id           = "${var.project}-${var.environment}"
-  engine               = "redis"
-  node_type            = var.node_type
-  num_cache_nodes      = var.node_count
-  parameter_group_name = var.parameter_group_name
-  engine_version       = var.engine_version
-  port                 = 6379
-  subnet_group_name    = aws_elasticache_subnet_group.main.name
-  apply_immediately    = true
-
+resource "aws_elasticache_replication_group" "main" {
+  replication_group_id       = "${var.project}-${var.environment}"
+  description                = "Collection of Redis cache clusters for ${var.project}-${var.environment}"
+  engine                     = "redis"
+  node_type                  = var.node_type
+  num_cache_clusters         = var.node_count
+  parameter_group_name       = var.parameter_group_name
+  engine_version             = var.engine_version
+  port                       = 6379
+  subnet_group_name          = aws_elasticache_subnet_group.main.name
+  at_rest_encryption_enabled = true
```

Comment:
great addtition!

let's also add `snapshot_retention_limit` and default it to `0`. Have it as a variable and mention in the readme that 0 is no backups. We need backups for some apps.